cmd/agent/common/helpers.go

```diff
@@ -49,3 +51,34 @@ func setupConfig(confFilePath string, configName string, withoutSecrets bool) er
 	}
 	return nil
 }
+
+// SetupSystemProbeConfig adds the system-probe.yaml file to the config object
+func SetupSystemProbeConfig(sysProbeConfFilePath string) error {
+	// set the paths where a config file is expected. Without this, if there is a system-probe.yaml
+	// in the default location it will overwrite whatever custom path is passed in
+	if len(sysProbeConfFilePath) != 0 {
+		// If they pass in the file directly, assume we should use it
+		if strings.HasSuffix(sysProbeConfFilePath, ".yaml") {
+			config.Datadog.SetConfigFile(sysProbeConfFilePath)
```

Comment:
won't that overwrite the default config path ? Will we have any conflict between the system-probe yaml and datadog.yaml ?

Reply:
Ahh, you're right - this will. I'm going to look into whether or not we can prevent this from happening, since I think for certain agents this will be a problem where we need use the config in other binaries and they call ConfigFileUsed()